src/ome_zarr_models/common/multiscales.py

```diff
@@ -207,43 +59,83 @@ def build(
             ),
         )
 
+    @field_validator("coordinateTransformations", mode="before")
+    def _ensure_scale_translation(
+        transforms_obj: object,
+    ) -> object:
+        """
+        Ensures that
+        - there are only 1 or 2 transforms.
+        - the first element is a scale transformation
+        - the second element, if present, is a translation transform
+        """
+        # This is used as a before validator - to help use, we use pydantic to first
+        # cast the input (which can in general anything) into a set of transformations.
+        # Then we check the transformations are valid.
+        #
+        # This is a bit convoluted, but we do it because the default pydantic error
+        # messages are a mess otherwise
 
-def _ensure_ordered_scales(datasets: list[Dataset]) -> list[Dataset]:
-    """
-    Make sure datasets are ordered from highests resolution to smallest.
-    """
-    scale_transforms = [d.coordinateTransformations[0] for d in datasets]
-    # Only handle scales given in metadata, not in files
-    scale_vector_transforms = [
-        t for t in scale_transforms if isinstance(t, VectorScale)
-    ]
-    scales = [s.scale for s in scale_vector_transforms]
-    for i in range(len(scales) - 1):
-        s1, s2 = scales[i], scales[i + 1]
-        is_ordered = all(s1[j] <= s2[j] for j in range(len(s1)))
-        if not is_ordered:
-            raise ValueError(
-                f"Dataset {i} has a lower resolution (scales = {s1}) "
-                f"than dataset {i+1} (scales = {s2})."
+        class Transforms(BaseModel):
+            transforms: list[Transform]
+
+        transforms = Transforms(transforms=transforms_obj).transforms
```

Comment:
this looks odd, what is this `Transforms` class for exactly?

Reply:
Reading the comment above, this is to parse the input as a list of `Transform`, which we need to do because this is a "before" validator. I'm not 100% why we don't just have this as an after validator so we don't need to bother with this though...